deepcell/image_generators/semantic.py

```diff
@@ -385,3 +375,360 @@ def random_transform(self, x, y=None, seed=None):
+
+            Default is 'nearest'. Points outside the boundaries of the input
+            are filled according to the given mode:
+                - 'constant': kkkkkkkk|abcd|kkkkkkkk (cval=k)
+                - 'nearest':  aaaaaaaa|abcd|dddddddd
+                - 'reflect':  abcddcba|abcd|dcbaabcd
+                - 'wrap':  abcdabcd|abcd|abcdabcd
+
+        cval: float or int, value used for points outside the boundaries
+            when fill_mode = "constant".
+        horizontal_flip: boolean, randomly flip inputs horizontally.
+        vertical_flip: boolean, randomly flip inputs vertically.
+        rescale: rescaling factor. Defaults to None. If None or 0, no rescaling
+            is applied, otherwise we multiply the data by the value provided
+            (before applying any other transformation).
+        preprocessing_function: function that will be implied on each input.
+            The function will run after the image is resized and augmented.
+            The function should take one argument:
+            one image (Numpy tensor with rank 3),
+            and should output a Numpy tensor with the same shape.
+
+        data_format: One of {"channels_first", "channels_last"}.
+
+            - "channels_last" mode means that the images should have shape
+              (samples, height, width, channels),
+            - "channels_first" mode means that the images should have shape
+              (samples, channels, height, width).
+            - It defaults to the image_data_format value found in your
+              Keras config file at "~/.keras/keras.json".
+            - If you never set it, then it will be "channels_last".
+
+        validation_split: float, fraction of images reserved for validation
+            (strictly between 0 and 1).
+    """
+    def __init__(self, **kwargs):
+        super(SemanticMovieGenerator, self).__init__(**kwargs)
+        # Change the axes for 5D data
+        if self.data_format == 'channels_first':
+            self.channel_axis = 1
+            self.row_axis = 3
+            self.col_axis = 4
+            self.time_axis = 2
+        if self.data_format == 'channels_last':
+            self.channel_axis = 4
+            self.row_axis = 2
+            self.col_axis = 3
+            self.time_axis = 1
+
+    def flow(self,
+             train_dict,
+             batch_size=1,
+             frames_per_batch=5,
+             transforms=['watershed-cont'],
+             transforms_kwargs={},
+             shuffle=True,
+             min_objects=3,
+             seed=None):
+        """Generates batches of augmented/normalized data with given arrays.
+
+        Args:
+            train_dict: dictionary of X and y tensors. Both should be rank 4.
+            batch_size: int (default: 1).
+            shuffle: boolean (default: True).
+            seed: int (default: None).
+
+        Returns:
+            An Iterator yielding tuples of (x, y) where x is a numpy array
```

Comment:
This applies for all the docstrings (I think), but sphinx picks up the type of the input from the docstring like:

```
train_dict (dict): dictionary of X and y tensors.
batch_size (int): the size of the batches.
shuffle (bool): whether or not to shuffle the data.
seed (int): random seed.

Returns:
    SemanticMovieIterator: Iterator yielding....
```

The Args use the type in parens like `(int)` and the Returns just uses the type and then a description.
Feel free to copy some of the docstrings from other generators as a template.